### DISCUSSION TITLE
Reference Architecture - Can we use a different primary region per environment?

### DISCUSSION THREAD

**Discussion by zackproser** [gruntwork-io/knowledge-base#216] [Help] [answered]
A customer asked: 
> We'd like our logs, security and shared accounts to use `us-east-1` as their primary region. However, dev, stage and prod should be deployed in `sa-east-1`. How can we define the different regions for use in different accounts?

**Comment by zackproser** [comment, marked as answer]
Unfortunately, we don’t currently support a mixed region Reference Architecture deployment. We’re able to provide the Ref Arch at a flat fee with a quick turnaround due to our standardization, which means the limited options in the form are the only things we can support in deployment.

Given your use case, I recommend deploying the Ref Arch in the us-east-1 region, and then migrating the app setup to sa-east-1 post deploy. 

The steps to do this are:

- Copy paste the us-east-1 folder into a new sa-east-1 folder in each of the environments dev, stage, and prod.
- Update the region.hcl file to point to sa-east-1
- Copy all secrets manager entries in us-east-1 into sa-east-1
- Update ARN references to secrets manager entries in the code to point to the newly copied secrets manager entries in sa-east-1.
- Run terragrunt run-all apply in the sa-east-1 folder.

Note that Gruntwork is happy to help out in support if you run into any issues going through this process.

**Comment by zackproser** [comment]
Credit for this solution goes to @yorinasub17

**Comment by moos3** [comment]
How are you suppose to run the terragrunt run-all apply locally? I dont believe iam admin users have the same permissions as the `ci-machine-user`.  @zackproser @yorinasub17 @pete0emerson can one of you chime in on this?

**Reply by yorinasub17** [reply]
In a standard Reference Architecture, you can create IAM users that have the ability to assume the full access IAM role to have admin privileges in the target account (described in https://github.com/gruntwork-io/terraform-aws-service-catalog/blob/master/examples/for-production/infrastructure-live/docs/02-authenticate.md#configure-other-iam-users).

In a CIS reference architecture, there is a special ops-admin role that provides the same permissions as the deploy runner, which can be configured in a similar manner (with the group being `_account.dev-ops-admin`).

Once an IAM user is configured in this way, they can assume the `allow-full-access-from-other-accounts` (or `allow-ops-admin-access-from-other-accounts` for CIS) IAM roles from the security account to get the necessary permissions.
